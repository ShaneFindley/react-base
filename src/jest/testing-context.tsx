import { createContext } from 'react'

export interface ComponentContextShape {
  value: string | undefined
  onClick?: (value: string) => void
}

export const ComponentContext = createContext<ComponentContextShape>({ value: 'none', onClick: (value: string) => {} })
