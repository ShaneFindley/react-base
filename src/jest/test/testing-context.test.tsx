import { render } from '@testing-library/react'
import { ComponentContext } from '../testing-context'

it('should display values 1', () => {
  const { container } = setupWithElement(<span>Here we go</span>)
  expect(container).toMatchSnapshot()
})

it('should display values 2', () => {
  const { container } = setupWithElement(<span>Here we go again</span>)
  expect(container).toMatchSnapshot()
})

it('should display values 3', () => {
  const { container } = setupWithElement(<span>Here we go again and again</span>)
  expect(container).toMatchSnapshot()
})

function setupWithElement (underTest: JSX.Element) {
  return render(<ComponentContext.Provider value={{ value: 'none', onClick: () => {} }}>{underTest}</ComponentContext.Provider>)
}
