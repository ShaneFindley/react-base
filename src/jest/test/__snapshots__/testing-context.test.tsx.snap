// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display values 1 1`] = `
<div>
  <span>
    Here we go
  </span>
</div>
`;

exports[`should display values 2 1`] = `
<div>
  <span>
    Here we go again
  </span>
</div>
`;

exports[`should display values 3 1`] = `
<div>
  <span>
    Here we go again and again
  </span>
</div>
`;
